eng/tools/lint-diff/src/processChanges.ts

```diff
@@ -356,31 +224,53 @@ export function getService(filePath: string): string {
 }
 
 /**
- * Given a root path and directory, find all files whose name ends with a given
- * string. (e.g. ".json")
- * @param rootPath Root path of starting directory
- * @param dir Starting directory
- * @param endsWith a file extension to search for
- * @returns A list of files with paths relative to rootPath
+ * Return true if the path contains "/examples/"
+ * @param path 
+ * @returns 
  */
-async function enumerateFiles(rootPath: string, dir: string, endsWith: string): Promise<string[]> {
-  let results: string[] = [];
-  let stack: string[] = [dir];
-
-  while (stack.length > 0) {
-    const currentDir = stack.pop()!;
-    const list = await readdir(join(rootPath, currentDir), { withFileTypes: true });
-
-    for (const file of list) {
-      if (file.isDirectory()) {
-        stack.push(join(currentDir, file.name));
-      } else {
-        if (file.name.endsWith(endsWith)) {
-          results.push(join(currentDir, file.name));
-        }
-      }
+const excludeExamples = (path: string) => path.includes("/examples/");
+
+/**
+ * Given a list of swagger files relatve to before and after root paths,
+ * return a set of swagger files that have changed. Changes can be directly in 
+ * the swagger file or in part of a referenced file that is included. Not all 
+ * changes to a file in a $ref will affect a given swagger file.
+ * @param beforeRoot 
+ * @param afterRoot 
+ * @param affectedSwaggerCandidates 
+ * @returns 
+ */
+export async function getChangedSwaggers(
+  beforeRoot: string,
+  afterRoot: string,
+  affectedSwaggerCandidates: Set<string>,
+) {
+  const affectedSwaggers = new Set<string>();
+
+  for (const swagger of affectedSwaggerCandidates) {
+    const beforeSwagger = join(beforeRoot, swagger);
+    if (!(await pathExists(beforeSwagger))) {
+      affectedSwaggers.add(swagger);
+      continue;
+    }
+
+    const afterSwagger = join(afterRoot, swagger);
+
+    // Using dereference which supports excluding $ref paths (in this case, examples)
+    const derefBefore = await $RefParser.dereference(
+      beforeSwagger, 
+      { dereference: { excludedPathMatcher: excludeExamples}},
+    );
+    const derefAfter = await $RefParser.dereference(
+      afterSwagger, 
+      { dereference: { excludedPathMatcher: excludeExamples}},
+    );
+
+    // Compare the dereferenced objects
+    if (!deepEqual(derefBefore, derefAfter)) {
```

Comment:
This comparison uses [deep-eql](https://www.npmjs.com/package/deep-eql) a zero-dependency deep equality checker that supports objects with cycles (dereferenced swagger might contain cycles).